packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js

```diff
@@ -727,7 +733,34 @@ RED.palette.editor = (function() {
         });
     }
 
-    var settingsPane;
+    /**
+     * Initializes the check for updates triggered every 30 minutes.
+     * This interval is based on the cumulative active window time.
+     */
+    function initCheckForUpdatesInterval() {
+        let activeTime = 0, lastActiveTime = Date.now();
+        const INTERVAL = 30 * 60;   // 30min
+
+        document.addEventListener("visibilitychange", function () {
+            if (document.visibilityState === "visible") {
+                lastActiveTime = Date.now();
+            } else {
+                activeTime += (Date.now() - lastActiveTime) / 1000;
+            }
+        });
+
+        setInterval(function () {
+            if (document.visibilityState === "visible") {
+                activeTime += (Date.now() - lastActiveTime) / 1000;
+                lastActiveTime = Date.now();
+
+                if (activeTime >= INTERVAL) {
+                    activeTime = 0;
+                    getSettingsPane();
```

Comment:
As mentioned in the other PR - we need a separate API for refreshing the catalogues if we choose to go ahead with this PR (I'm not sold on the idea as being that useful given the typical usage of the editor).

The key point being - `getSettingsPane()` will cause the browser to generate the whole DOM needed for the palette manager; that's a lot of work that should only be done when we want to show the palette manager. It should not be done as part of the background check to see if updates are available.